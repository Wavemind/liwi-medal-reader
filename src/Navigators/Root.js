/**
 * Used to navigating without the navigation prop
 * @see https://reactnavigation.org/docs/navigating-without-navigation-prop/
 *
 * You can add other navigation functions that you need and export them
 */
import * as React from 'react'
import { CommonActions } from '@react-navigation/native'

export const navigationRef = React.createRef()

export function navigate(name, params) {
  navigationRef.current?.navigate(name, params)
}

export function navigateAndReset(routes = [], index = 0) {
  navigationRef.current?.dispatch(
    CommonActions.reset({
      index,
      routes,
    }),
  )
}

export function navigateAndSimpleReset(name, params = {}, index = 0) {
  navigationRef.current?.dispatch(
    CommonActions.reset({
      index,
      routes: [{ name, params: { ...params } }],
    }),
  )
}
/**
 * Use for redirection after PIN
 */
export function navigateToMedicalCase(medicalCase) {
  navigationRef.current?.dispatch(
    CommonActions.reset({
      actions: [
        navigationRef.current?.navigate('Home', {
          screen: 'StageWrapper',
          params: {
            stageIndex: medicalCase.advancement.stage,
            stepIndex: medicalCase.advancement.step,
          },
        }),
      ],
    }),
  )
}

export function navigateToStage(stageIndex, stepIndex = 0) {
  navigationRef.current?.dispatch(
    CommonActions.reset({
      index: 0,
      routes: [{ name: 'StageWrapper', params: { stageIndex, stepIndex } }],
    }),
  )
}

export function navigateNestedAndSimpleReset(name, nestedName, index = 0) {
  navigationRef.current?.dispatch(
    CommonActions.reset({
      index,
      routes: [{ name: name, state: { routes: [{ name: nestedName }] } }],
    }),
  )
}
