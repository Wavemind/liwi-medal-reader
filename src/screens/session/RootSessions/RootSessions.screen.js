// @flow

import * as React from 'react';
import { Button, Container, Content, Icon, Text, View } from 'native-base';
import { AppState, Image, ImageBackground, ScrollView } from 'react-native';
import { postDeviceInfo } from '../../../engine/api/Http';
import { styles } from './RootSessions.styles';
import Sessions from '../../../components/Sessions';
import type { I18nTypes } from '../../../utils/i18n';
import { NavigationScreenProps } from 'react-navigation';
import { clearLocalStorage } from 'engine/api/LocalStorage';
import { Fragment } from 'react';

type Props = NavigationScreenProps & {
  logged: boolean,
  t: I18nTypes,
  app: {
    authentication: (email: string, password: string) => void,
  },
};

type State = {};

export default class RootSessions extends React.Component<Props, State> {
  state = {};

  render() {
    const {
      navigation,
      t,
      app: { isConnected },
      sessions,
    } = this.props;

    return (
      <Fragment>
        <View style={styles.header}>
          <ImageBackground
            source={require('../../../../assets/images/tropical.png')}
            style={styles.backgroundImage}
          >
            <View style={styles.contentLogo}>
              <Image
                source={require('../../../../assets/images/icon.png')}
                style={styles.image}
                resizeMode={'contain'}
              />
            </View>
          </ImageBackground>
        </View>
        <Container style={styles.container}>
          <Content contentContainerStyle={styles.content}>
            <Sessions navigation={navigation} />
            <Text>Tools</Text>
            <Button
              disabled={!isConnected}
              iconLeft
              style={!isConnected ? { backgroundColor: '#69635f' } : {}}
              blue
              onPress={() => navigation.push('NewSession')}
            >
              {!isConnected ? (
                <Icon type={'MaterialCommunityIcons'} name="lan-disconnect" />
              ) : (
                <Icon type={'Feather'} name="user-plus" />
              )}

              <Text> {t('add_account')} </Text>
            </Button>
            <Button
              iconLeft
              blue
              onPress={() => {
                postDeviceInfo();
              }}
            >
              <Icon type={'Octicons'} name="device-mobile" />
              <Text>{t('send_device_info')}</Text>
            </Button>
            <Button
              iconLeft
              blue
              onPress={async () => {
                await clearLocalStorage('sessions');
                await sessions.initContext();
              }}
            >
              <Icon type={'MaterialCommunityIcons'} name="delete-forever" />
              <Text>{t('clear_sessions')}</Text>
            </Button>
          </Content>
        </Container>
      </Fragment>
    );
  }
}
