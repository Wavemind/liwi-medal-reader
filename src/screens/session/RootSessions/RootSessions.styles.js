import { Dimensions, KeyboardAvoidingView, StyleSheet } from 'react-native';
import { screenWidth, screenHeight } from '../../../utils/constants';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
  header: {
    alignItems: 'center',
    backgroundColor: 'rgba(0, 0, 0, 0.0)',
  },
  contentLogo: {
    alignItems: 'center',
  },
  backgroundImage: {
    width: screenWidth,
    backgroundColor: 'rgba(0, 0, 0, 0.0)',
  },
  content: {
    flex: 1,
    alignItems: 'center',
  },
  image: {
    width: 100,
  },
});
