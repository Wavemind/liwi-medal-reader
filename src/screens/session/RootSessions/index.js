import RootSessions from './RootSessions.screen';
import { withNamespaces } from 'react-i18next';
import { withApplication } from '../../../engine/contexts/Application.context';
import { withSessions } from 'engine/contexts/Sessions.context';

export default withApplication(
  withSessions(withNamespaces(['login'])(RootSessions))
);
