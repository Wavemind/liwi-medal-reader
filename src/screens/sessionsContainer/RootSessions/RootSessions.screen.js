// @flow

import * as React from 'react';
import { Fragment } from 'react';
import { Button, Container, Content, Icon, Text, View } from 'native-base';
import { AppState, Image, ImageBackground, ScrollView } from 'react-native';
import { post } from '../../../engine/api/Http';
import { styles } from './RootSessions.styles';
import Sessions from '../../../components/Sessions';
import type { I18nTypes } from '../../../utils/i18n';
import { NavigationScreenProps } from 'react-navigation';
import { clearMedicalCases, clearSessions } from 'engine/api/LocalStorage';
import { GetDeviceInformations } from '../../../engine/api/Device';

type Props = NavigationScreenProps & {
  logged: boolean,
  t: I18nTypes,
  app: {
    authentication: (email: string, password: string) => void,
  },
};

type State = {};

export default class RootSessions extends React.Component<Props, State> {
  state = {};

  render() {
    const {
      navigation,
      t,
      app: { isConnected },
      sessions,
    } = this.props;

    return (
      <Fragment>
        <View style={styles.header}>
          <ImageBackground
            source={require('../../../../assets/images/tropical.png')}
            style={styles.backgroundImage}
          >
            <View style={styles.contentLogo}>
              <Image
                source={require('../../../../assets/images/icon.png')}
                style={styles.image}
                resizeMode={'contain'}
              />
            </View>
          </ImageBackground>
        </View>
        <Container style={styles.container}>
          <Content contentContainerStyle={styles.content}>
            <Sessions navigation={navigation} />
            <Text>Tools</Text>
            <Button
              disabled={!isConnected}
              iconLeft
              style={!isConnected ? { backgroundColor: '#69635f' } : {}}
              blue
              onPress={() => navigation.push('NewSession')}
            >
              {!isConnected ? (
                <Icon type={'MaterialCommunityIcons'} name="lan-disconnect" />
              ) : (
                <Icon type={'MaterialCommunityIcons'} name="lan-connect" />
              )}

              <Text> {t('add_account')} </Text>
            </Button>
            <Button
              iconLeft
              blue
              onPress={async () => {
                await GetDeviceInformations((deviceInfo) => {
                  post('activities', deviceInfo);
                });
              }}
            >
              <Icon type={'Octicons'} name="device-mobile" />
              <Text>{t('send_device_info')}</Text>
            </Button>
            <Button
              iconLeft
              blue
              onPress={async () => {
                await clearSessions('sessions');
                await sessions.initContext();
              }}
            >
              <Icon type={'MaterialCommunityIcons'} name="delete-forever" />
              <Text>{t('clear_sessions')}</Text>
            </Button>
            <Button
              iconLeft
              blue
              onPress={async () => {
                await clearMedicalCases();
              }}
            >
              <Icon type={'MaterialCommunityIcons'} name="delete-forever" />
              <Text>Vider les cas m√©dicals</Text>
            </Button>
          </Content>
        </Container>
      </Fragment>
    );
  }
}
