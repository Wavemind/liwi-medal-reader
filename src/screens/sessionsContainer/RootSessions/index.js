import RootSessions from './RootSessions.screen';
import { withApplication } from '../../../engine/contexts/Application.context';
import { withSessions } from 'engine/contexts/Sessions.context';
import { actions } from '../../../../frontend_service/actions/types.actions';
import { connect } from 'react-redux';


const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    clear: () => dispatch({ type: actions.MC_CLEAR }),
  };
};

const mapStateToProps = (medicalCase, ownProps) => {
  return {
    medicalCase,
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withApplication(withSessions(RootSessions)));
