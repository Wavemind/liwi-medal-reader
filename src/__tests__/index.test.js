// __tests__/Intro-test.js
import * as React from 'react';
import renderer from 'react-test-renderer';
import Root from 'index';
import RootLayout from 'template/Layout.template';
import RootSessions from 'screens/session/RootSessions';

test('render Root', () => {
  const tree = renderer.create(<Root />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render Root Layout', () => {
  const tree = renderer.create(<RootLayout />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('render Root Sessions', () => {
  const tree = renderer.create(<RootSessions />).toJSON();
  expect(tree).toMatchSnapshot();
});
