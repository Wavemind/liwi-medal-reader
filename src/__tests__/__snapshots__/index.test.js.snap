// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Root 1`] = `null`;

exports[`render Root Sessions 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.0)",
      },
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.0)",
        "width": 750,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "width": 100,
          }
        }
      />
    </View>
  </View>
</View>
`;
